pkg/install/process.go

```diff
@@ -16,135 +16,62 @@ limitations under the License.
 package install
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
-	"os/exec"
-	"runtime"
-	"strings"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
 
-	"github.com/k0sproject/k0s/internal/util"
-	"github.com/mitchellh/go-ps"
-	"gopkg.in/yaml.v2"
+	config "github.com/k0sproject/k0s/pkg/apis/v1beta1"
+	"github.com/k0sproject/k0s/pkg/constant"
 )
 
 type K0sStatus struct {
-	Version  string
-	Pid      int
-	PPid     int
-	Role     string
-	SysInit  string
-	StubFile string
-	Output   string
+	Version       string
+	Pid           int
+	PPid          int
+	Role          string
+	SysInit       string
+	StubFile      string
+	Output        string
+	Workloads     bool
+	Args          []string
+	ClusterConfig *config.ClusterConfig
+	K0sVars       constant.CfgVars
 }
 
-func GetPid() (status *K0sStatus, err error) {
-	pid, ppid, err := getProcessID()
-	if err == nil && pid != nil {
-		status = &K0sStatus{
-			Pid:  *pid,
-			PPid: *ppid,
-		}
-		return status, nil
-	}
-	return &K0sStatus{}, nil
-}
-
-func getProcessID() (pid *int, ppid *int, err error) {
-	processList, err := ps.Processes()
-	if err != nil {
-		return nil, nil, err
-	}
-
-	for _, p := range processList {
-		if p.Executable() == "k0s" && hasChildren(p.Pid(), processList) {
-			pid := p.Pid()
-			ppid := p.PPid()
-			return &pid, &ppid, nil
-		}
-	}
-	return nil, nil, nil
-}
-
-func hasChildren(pid int, processes []ps.Process) bool {
-	for _, p := range processes {
-		if p.PPid() == pid {
-			return true
-		}
-	}
-	return false
-}
+func GetStatusInfo(runDir string, socket string) (status *K0sStatus, err error) {
+	status = &K0sStatus{}
 
-func GetRoleByPID(pid int) (role string, err error) {
-	if runtime.GOOS == "windows" {
-		return "worker", nil
+	address := filepath.Join(runDir, socket)
+	if _, err := os.Stat(address); err != nil {
+		return nil, fmt.Errorf("%s does not exist", address)
 	}
-
-	var raw []byte
-	if raw, err = ioutil.ReadFile(fmt.Sprintf("/proc/%d/cmdline", pid)); err != nil {
-		return "", err
-	}
-	cmdln := string(raw)
-	if strings.Contains(cmdln, "enable-worker") || strings.Contains(cmdln, "single") {
-		return "controller+worker", nil
-	} else if strings.Contains(cmdln, "controller") {
-		return "controller", nil
-	} else if strings.Contains(cmdln, "worker") {
-		return "worker", nil
-	} else if strings.Contains(cmdln, "server") {
-		return "controller", nil
+	httpc := http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", address)
+			},
+		},
 	}
-	return "", fmt.Errorf("k0s role is not found")
-}
 
-func (s K0sStatus) GetK0sVersion() (string, error) {
-	cmd := fmt.Sprintf("/proc/%d/exe", s.Pid)
-	stdout, err := exec.Command(cmd, "version").Output()
+	response, err := httpc.Get("http:/" + address)
```

Comment:
Whats with the single `/`?

Reply:
I tried the address string is file path so it has a slash at the beginning

Reply:
The transport/connection is setup with `unix://<address>`, and your handler doesn't require any paths, so shouldn't `httpc.Get("/")` just work here?

Reply:
`httpc.Get("/")` won't work. I replaced it with `https.Get("http://localhost")` so we don't have to concatenate strings.